### ISSUE TITLE
Invalid value "2017-02-20"  and  ly_ctx_new() failed  when timezone is Europe/Dublin on Debian

### ISSUE THREAD

**Issue by jfbucas** [CESNET/libyang#1889]
Using [Frr project](https://github.com/FRRouting/frr) on Debian with deb package **frr 8.3-0~deb10u1** which installs  [libyang2 2.0.7-1](https://deb.frrouting.org/frr/pool/frr-stable/liby/libyang2/libyang2_2.0.7-1~deb11u1_amd64.deb)

When I start **/usr/lib/frr/watchfrr**, I get the error:
```
2022/07/26 14:07:57 WATCHFRR: libyang: Invalid value "2017-02-20" of "revision". (Line number 32.)
2022/07/26 14:07:57 WATCHFRR: [WH45S-QHX34][EC 100663345] yang_init: ly_ctx_new() failed
```

It seems to happens as libyang2 tries to read /etc/localtime

Changing the timezone to Europe/London or Europe/Belfast (or any other timezone) fixes this problem.

**Comment by michalvasko** [comment]
*libyang* 2.0.7 is more than a year old when version 2 had just been shortly available publicly, so there were many problems. You will have to update it to fix this problem.

**Comment by tlsalmin** [comment]
Still same issue persists with build from 2.0.231. Only line changes to 38:

libyang: Invalid value "2017-02-20" of "revision". (Line number 38.)

**Comment by michalvasko** [comment]
Okay, it should be fixed and this is then a duplicate of #1896.
